### ISSUE TITLE
ShadowingOuterLocalVariable and UnusedLocalVariable triggered by Regexp with named captures

### ISSUE THREAD

**Issue by badal** [rubocop/rubocop#332]
2 errors occurred:
An error occurred while ShadowingOuterLocalVariable cop was inspecting 
...
An error occurred while UnusedLocalVariable cop was inspecting ...

0.9.0 (using Parser 2.0.0.beta9, running on ruby 1.9.3 i386-mingw32)

Both errors are triggered by the following line :

```
   if /(?<token>\w+)(?<setter>=?)/ =~ name.to_s && token_names.include?(token)
```
